synapse/config/_base.py

```diff
@@ -704,12 +713,33 @@ def load_or_generate_config(
             == _ConfigGenerateMode.GENERATE_EVERYTHING_AND_EXIT
         ):
             if config_args.report_stats is None:
-                parser.error(
-                    "Please specify either --report-stats=yes or --report-stats=no\n\n"
-                    + MISSING_REPORT_STATS_SPIEL
+                report_stats_file = cls.pop_config_by_name(
+                    config_files, "report_stats.yaml"
                 )
+                if report_stats_file:
+                    with open(report_stats_file) as f:
+                        config_args.report_stats = yaml.safe_load(f)["report_stats"]
+                else:
+                    parser.error(
+                        "Please specify either --report-stats=yes or --report-stats=no\n\n"
+                        + MISSING_REPORT_STATS_SPIEL
+                    )
```

Comment:
Setting aside for now the fact that we're hardcoding some very debian-package-specific magic into the main code here:

It's worth noting that, at present, you can `dpkg-reconfigure matrix-synapse-py3` and change the `report_stats` setting. That will be written back to `/etc/matrix-synapse/conf.d/report_stats.yaml` so that it is picked up by synapse on the next restart.

It looks to me as if this change will put the `report_stats` setting into `/etc/matrix-synapse/homeserver.yaml` on first run, where it will never be changed thereafter - so users can no longer `dpkg-reconfigure` this setting.

Reply:
That's a good point, I'll think about how to address this.